### PR TITLE
Install Lightning CSS dependency and process rollup-plugin CSS using …

### PR DESCRIPTION
**Pull request by Prakshal-Jain**
## What changed / motivation ?
Using Lightning CSS to process the output of `stylexBabelPlugin.processStylexRules`, which outputs CSS rules. As a starting point, I've installed the package as dependency, and currently only being used in `rollup-plugin`.

## Additional Context
Using Lightning CSS `transform` function helps with:
- optimizing CSS for performance by minifying and reducing file sizes
- ensuring cross-browser compatibility through automatic prefixing and polyfilling.

Screenshots, Tests, Anything Else
The tests `packages/rollup-plugin/__tests__/index-test.js` is updated with the new output from `transform()` function, and successfully passes.
image

## Resources
1. Lightning CSS Documentation: https://lightningcss.dev/docs.html
2. Github: https://github.com/parcel-bundler/lightningcss

### CODE REVIEW

File: package.json
Comment (nmn) [inline package.json:54]: This should be added as dependency to the `rollup-plugin` and not the monorepo itself.

File: packages/rollup-plugin/__tests__/index-test.js
Comment (nmn) [inline packages/rollup-plugin/__tests__/index-test.js:65]: Look into the options for lightningcss so we format CSS in a more minimal formay.

Comment (nmn) [reply packages/rollup-plugin/__tests__/index-test.js:65]: I couldn't find such an option, so maybe it's OK to leave it expanded.

File: packages/rollup-plugin/src/index.js
Comment (nmn) [inline packages/rollup-plugin/src/index.js:70]: This is a good start, but you must enable the option for vendor prefixing for this to be useful:

https://lightningcss.dev/transpilation.html#browser-targets

**Review by nmn — CHANGES_REQUESTED** [review]
Good start. A few more changes needed before it can be merged.

File: packages/rollup-plugin/__tests__/index-test.js
Comment (Prakshal-Jain) [reply packages/rollup-plugin/__tests__/index-test.js:65]: We can use `minify: true` option for this.
For example:
```
import { transform } from 'lightningcss';

let { code, map } = transform({
  filename: 'style.css',
  code: Buffer.from('.foo { color: red }'),
  minify: true,
  sourceMap: true
});
```

Documentation: https://lightningcss.dev/docs.html

Comment (nmn) [reply packages/rollup-plugin/__tests__/index-test.js]: @Prakshal-Jain `minify: true` is more aggressive than we want. It will also get rid of all the line breaks and make the CSS generally unreadable. It should not be turned on by default.

However, the `rollup-plugin` should be updated accept props that are passed into `lightningcss` and it should be possible to pass in `minify: true` within `lightningcssArgs: {...}` by the developer.

File: packages/rollup-plugin/__tests__/index-test.js
Comment (nmn) [inline packages/rollup-plugin/__tests__/index-test.js:65]: Yeah this is worse than adding even more whitespace.

File: packages/rollup-plugin/src/index.js
Comment (nmn) [inline packages/rollup-plugin/src/index.js:67]: remove `minify: true`, but enable "syntax lowering". You can use a default value for `browsers` but the user should be able to override it.

**Review by nmn — CHANGES_REQUESTED** [review]
Nice work getting most of the signals to be green.

File: packages/rollup-plugin/__tests__/index-test.js
Comment (Prakshal-Jain) [reply packages/rollup-plugin/__tests__/index-test.js]: Okay. Let me disable this for now. I'll add the lightningcssArgs in the next commit to add support for this argument.

File: packages/rollup-plugin/src/index.js
Comment (Prakshal-Jain) [reply packages/rollup-plugin/src/index.js]: For the vendor list, would you recommend doing this:
```
let targets = browserslistToTargets(browserslist('>= 0.25%'));
```

File: packages/rollup-plugin/src/index.js
Comment (Prakshal-Jain) [reply packages/rollup-plugin/src/index.js]: I think "syntax lowering" will be enabled when we add the supported browsers in the vendor list
https://lightningcss.dev/transpilation.html#syntax-lowering

File: packages/rollup-plugin/__tests__/index-test.js
Comment (Prakshal-Jain) [reply packages/rollup-plugin/__tests__/index-test.js]: Removed this in the newer commit. Back to the original format

File: packages/rollup-plugin/__tests__/index-test.js
Comment (Prakshal-Jain) [reply packages/rollup-plugin/__tests__/index-test.js:65]: Resolved. Back to original format.

File: packages/rollup-plugin/src/index.js
Comment (nmn) [reply packages/rollup-plugin/src/index.js]: Yes. That's what I'm saying. You should choose a good default list for browsers to support. Perhaps `>0.5%` like the example in the docs? But then users should be able to customise it with their own list of supported browsers.

File: packages/rollup-plugin/src/index.js
Comment (Prakshal-Jain) [inline packages/rollup-plugin/src/index.js:30]: @nmn We currently need to define this explicitly because Flow complains if I:
```
import type { TransformOptions } from 'lightningcss';
```

and use it like:
```
export type PluginOptions = $ReadOnly<{
  ...Partial<Options>,
  fileName?: string,
  babelConfig?: $ReadOnly<{
    plugins?: $ReadOnlyArray<PluginItem>,
    presets?: $ReadOnlyArray<PluginItem>,
  }>,
  useCSSLayers?: boolean,
  lightningcssOptions?: TransformOptions<{}>,      <--------Using the type
  ...
}>;
```


Do you have any suggestions to do this more elegantly?

**Comment by nmn** [comment]
The changes are starting look good. Work on fixing the signals next. Run ESLint etc locally.

File: packages/rollup-plugin/src/index.js
Comment (Prakshal-Jain) [reply packages/rollup-plugin/src/index.js]: Synced. Setting this to >1.0

**Comment by necolas** [comment]
@Prakshal-Jain Please don't merge branches, especially if they haven't been approved yet. You'll have to fix the lint errors in a follow up

**Pull request merged**
